src/swarm/neo/authentication/CredentialsFile.d

```diff
@@ -38,275 +38,219 @@
+import ocean.io.device.File;
+import ocean.net.util.QueryParams;
+import ocean.text.convert.Format;
 
 extern (C) private int g_ascii_xdigit_value ( char c ); // glib-2.0
 
-/******************************************************************************/
+deprecated("Replace imports of CredentialsFile with Credentials in "
+    "swarm.neo.authentication.NodeCredentials")
+public alias NodeCred.Credentials CredentialsFile;
 
-class CredentialsFile
-{
-    import swarm.neo.authentication.Credentials;
-    import HmacDef = swarm.neo.authentication.HmacDef;
-
-    import ocean.io.device.File;
+/***************************************************************************
 
-    import ocean.net.util.QueryParams;
+    Updates the credentials from the file, and changes this instance to
+    refer to the updated credentials on success. On error this instance will
+    keep referring to the same credentials.
 
-    import ocean.text.convert.Format;
+    Params:
+        filepath = the path of the file containing content
 
-    /***************************************************************************
+    Returns:
+        the authorisation keys by name
 
-        The credentials registry. A pointer to it can be obtained by
-        `credentials()`, it changes when `update()` is called.
+    Throws:
+        - IOException on file I/O error.
+        - ParseException on invalid file size or content; that is, if
+          - the file size is greater than Credentials.LengthLimit.File,
+          - a name
+            - is empty (zero length),
+            - is longer than Credentials.LengthLimit.File,
+            - contains non-graph characters,
+          - a key
+            - is empty (zero length),
+            - is longer than Credentials.LengthLimit.File * 2,
+            - has an odd (not even) length,
+            - contains non-hexadecimal digits.
 
-    ***************************************************************************/
+***************************************************************************/
 
-    private HmacDef.Key[istring] credentials_;
-
-    /***************************************************************************
-
-        The credentials file path.
-
-    ***************************************************************************/
-
-    private istring filepath;
-
-    /***************************************************************************
-
-        Constructor, reads the credentials from the file (by calling
-        `update()`).
-
-        Params:
-            filepath = the credentials file path
-
-        Throws:
-            See `update()`.
+public HmacDef.Key[istring] parse ( cstring filepath )
+{
+    scope file = new File(filepath, File.ReadExisting);
+    scope ( exit ) file.close;
 
-    ***************************************************************************/
+    size_t file_length = file.length;
 
-    public this ( istring filepath )
+    if (file_length > CredDef.LengthLimit.File)
```

Comment:
Yeah, I was also surprised to find this limit. It's a simplistic implementation that we can extend at some point, if needed.